### PR TITLE
Adderss management and toast notifications

### PR DESCRIPTION
**Pull request by MaruthiPothuganti**
- Users Can Add, Update, Delete and Edit the Addresses.
- Users can view the changes made in the app while navigating as a toast Notification.
- User Log persistence even during page refresh.

### CODE REVIEW

**Review by shashikantmaurya0007 — APPROVED** [review]
Cod looks good to be merged

File: src/Components/Address/Address.jsx [hunk 1]
```diff
@@ -1,5 +1,78 @@
...
+import { useAddress } from "../../Context/AddressContext";
+import { AddressModal } from "../AddressModal/AddressModal";
 
 export const Address = () => {
-  return <div>Address are here</div>;
+  const { addresses, removeAddress } = useAddress();
+  const [isAddressModalOpen, setIsAddressModal] = useState(false);
+
+  const initialAddressState = {
+    // _id: null,
```
Comment (anjali1102) [inline src/Components/Address/Address.jsx]: remove

File: src/App.css [hunk 2]
```diff
@@ -19,3 +19,7 @@ font-family: 'Poppins', sans-serif;
     --primary-bg-color: rgb(212, 212, 212);
     --secondary-bg-color: #1f2937;
 }
+
+/* .toast{
+    font-size: 1.5rem;
+} */
```
Comment (anjali1102) [inline src/App.css]: also this

**Review by anjali1102 — APPROVED** [review]

**Review by dev-enforced — APPROVED** [review]
The code is very well structured. Would look onto it to implement the address feature in my ecom too!! Good work Maruthi!! One error was coming up while viewing the cart page. You can have a look at this screenshot
image

This is what I did to get the error:
1) The address which was provided earlier in the profile section I removed the default tag from it and did not have any other address followed by that
2) I went to the cart page so no default address was there then the error will come up

File: src/App.css (hunk 2 above)
Comment (MaruthiPothuganti) [reply src/App.css]: Thank you for pointing it out. Will cleanup the commented code.

**Comment by MaruthiPothuganti** [comment]
> The code is very well structured. Would look onto it to implement the address feature in my ecom too!! Good work Maruthi!! One error was coming up while viewing the cart page. You can have a look at this screenshot image

Thankyou Roshan. Will look into that

**Pull request merged**